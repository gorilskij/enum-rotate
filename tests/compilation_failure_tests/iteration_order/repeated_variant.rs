use derive_enum_rotate::EnumRotate;

#[derive(EnumRotate)]
#[iteration_order(A, B, A)]
enum Enum {
    A, B, C
}
