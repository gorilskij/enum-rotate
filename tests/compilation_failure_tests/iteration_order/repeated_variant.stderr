error: Variant C not covered
 --> tests/compilation_failure_tests/iteration_order/repeated_variant.rs:4:18
  |
4 | #[iteration_order(A, B, A)]
  |                  ^^^^^^^^^
  |
  = note: Each variant of `Enum` should appear exactly once in the iteration order

error[E0601]: `main` function not found in crate `$CRATE`
 --> tests/compilation_failure_tests/iteration_order/repeated_variant.rs:7:2
  |
7 | }
  |  ^ consider adding a `main` function to `$DIR/tests/compilation_failure_tests/iteration_order/repeated_variant.rs`
